beacon/client/std-http-client.go

```diff
@@ -6,9 +6,11 @@ type StandardHttpClient struct {
 	*StandardClient
 }
 
-// Create a new client instance
-func NewStandardHttpClient(providerAddress string, timeout time.Duration) *StandardHttpClient {
-	provider := NewBeaconHttpProvider(providerAddress, timeout)
+// Create a new client instance.
+// Most calls will use the fast timeout, but queries to validator status will use the slow timeout since they can be very large.
+// Set a timeout of 0 to disable it.
+func NewStandardHttpClient(providerAddress string, fastTimeout time.Duration, slowTimeout time.Duration) *StandardHttpClient {
```

Comment:
I'd add a `type StandardHttpClientOpts struct` with two time.Duration fields. if non-0 they can be used as the defaults to context.WithTImeout. If 0, default to something sensible.

`NewStandardHttpClient("http://blah", nil)` would just use all defaults.